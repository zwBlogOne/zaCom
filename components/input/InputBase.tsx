import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BaseInputBaseProps } from './PropsType';
import Icon from '../icon';

export interface InputBaseProps extends BaseInputBaseProps {
  prefixCls?: string;
  className?: string;
}

export default class InputBase extends PureComponent<InputBaseProps, any> {
  static defaultProps = {
    prefixCls: 'za-input',
    disabled: false,
    type: 'text',
    clearable: true,
  };

  private input;
  private onBlurTimeout;
  private blurFromClear;

  constructor(props) {
    super(props);
    this.state = {
      focused: props.focused || false,
      value: props.defaultValue || props.value || '',
    };
  }

  componentDidMount() {
    if (this.props.autoFocus || this.state.focused) {
      this.input.focus();
    }
  }

  componentWillReceiveProps(nextProps) {
    const { value } = this.state;
    if ('focused' in nextProps) {
      this.setState({
        focused: nextProps.focused,
      });
    }

    if ('value' in nextProps && value !== nextProps.value ) {
      this.setState({
        value: nextProps.value,
      });
    }
  }

  componentWillUnmount() {
    if (this.onBlurTimeout) {
      clearTimeout(this.onBlurTimeout);
      this.onBlurTimeout = null;
    }
  }

  onFocus = (e) => {
    const value = e.target.value;
    if (!('focused' in this.props)) {
      this.setState({
        focused: true,
      });
    }

    const { onFocus } = this.props;
    if (typeof onFocus === 'function') {
      onFocus(value);
    }
  }

  onBlur = (e) => {
    const { onBlur } = this.props;
    const value = e.target.value;
    this.onBlurTimeout = setTimeout(() => {
      if (!this.blurFromClear && document.activeElement !== this.input) {
        this.setState({
          focused: false,
        });

        if (typeof onBlur === 'function') {
          onBlur(value);
        }
      }
      this.blurFromClear = false;
    }, 0);
  }

  handleComposition(e) {
    if (e.type === 'compositionstart') {
      this.setState({
        isOnComposition: true,
      });
      if (this.props.onCompositionStart) {
        this.props.onCompositionStart(e);
      }
    }

    if (e.type === 'compositionupdate') {
      if (this.props.onCompositionUpdate) {
        this.props.onCompositionUpdate(e);
      }
    }

    if (e.type === 'compositionend') {
      // composition is end
      this.setState({
        isOnComposition: false,
      });
      const value = e.target.value;
      if (this.props.onCompositionEnd) {
        this.props.onCompositionEnd(e);
      }
      if (this.props.onChange) {
        this.props.onChange(value);
      }
    }
  }

  onChange = (e) => {
    const { onChange } = this.props;
    if (!this.state.focused) {
      this.setState({
        focused: true,
      });
    }
    this.setState({
      value: e.target.value,
    });
    if (onChange) {
      onChange(e.target.value);
    }
  }

  onClear() {
    this.blurFromClear = true;
    const { value } = this.state;
    this.setState({
      value: '',
    });
    if (!this.state.isOnComposition) {
      this.focus();
    }
    if (this.props.onClear) {
      this.props.onClear(value);
    }
  }

  focus() {
    this.input.focus();
  }

  blur() {
    this.input.blur();
  }

  render() {
    const { prefixCls, className, disabled, onClear, clearable, type, ...others } = this.props;
    const { value, focused } = this.state;
    const cls = classnames(prefixCls, `${prefixCls}-${type}`, className, {
      disabled,
      clearable,
    });
    const clearCls = classnames(`${prefixCls}-clear`, {
      [`${prefixCls}-clear-show`]: !!(focused && value && value.length > 0),
    });
    return (
      <div className={cls}>
        <input
          {...others}
          autoComplete="off"
          ref={(ele) => { this.input = ele; }}
          type={type}
          disabled={disabled}
          value={value}
          onChange={this.onChange}
          onBlur={this.onBlur}
          onFocus={this.onFocus}
          onCompositionStart={(e) => { this.handleComposition(e); }}
          onCompositionUpdate={(e) => { this.handleComposition(e); }}
          onCompositionEnd={(e) => { this.handleComposition(e); }}
        />
        {clearable && <Icon type="wrong-round-fill" className={clearCls} onClick={() => { this.onClear(); }} />}
      </div>
    );
  }
}
