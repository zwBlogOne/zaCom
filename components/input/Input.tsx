import React, { PureComponent } from 'react';
import { BaseInputProps } from './PropsType';
import InputNumber from './InputNumber';
import InputBase from './InputBase';
import InputTextarea from './InputTextarea';

export interface InputProps extends BaseInputProps {
}

export default class Input extends PureComponent<BaseInputProps, {}> {
  static defaultProps = {
    type: 'text',
  };

  private input;
  private inputNumber;

  constructor(props) {
    super(props);
  }

  focus() {
    if (this.inputNumber) {
      this.inputNumber.focus();
    }
  }

  blur() {
    if (this.inputNumber) {
      this.inputNumber.blur();
    }
  }

  render() {
    const { type, ...others } = this.props;
    switch (type) {
      case 'idcard':
      case 'price':
      case 'number':
        return <InputNumber ref={ele => (this.inputNumber = ele)} {...others} type={type} />;

      case 'textarea':
        return <InputTextarea ref={ele => (this.input = ele)} {...others} />;

      default:
        return <InputBase ref={ele => (this.input = ele)} {...others} type={type} />;
    }
  }
}
