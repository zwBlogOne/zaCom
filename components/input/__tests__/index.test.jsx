import React from 'react';
import { render, mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../index';
import InputBase from '../InputBase';


describe('Input', () => {
  it('renders correctly', () => {
    const wrapper = render(<Input type="text" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('showLength', () => {
    const wrapper = render(<Input showLength maxLength={100} type="textarea" rows={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders onFocus called correctly', () => {
    const onFocus = jest.fn();
    const wrapper = mount(
      <Input onFocus={onFocus} />
    );
    wrapper.find('input[type="text"]').simulate('focus');
    expect(onFocus).toBeCalled();
    // expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders onBlur called correctly', () => {
    const wrapper = mount(<InputBase />);
    const spy = jest.spyOn(wrapper.instance(), 'onBlur');
    wrapper.find('input').simulate('focus');
    wrapper.find('input').simulate('blur');
    expect(spy).toHaveBeenCalled();
  });

  it('renders onClear called correctly', () => {
    const onClear = jest.fn();
    const wrapper = mount(
      <Input
        onClear={onClear}
        clearable
      />
    );

    const input = wrapper.find('input[type="text"]');
    input.simulate('change', { target: { value: 'My new value' } });
    wrapper.find('i.za-input-clear').simulate('click');
    expect(onClear).toHaveBeenCalled();
    expect(input.instance().value).toEqual('');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Input.Textarea', () => {
  it('renders correctly', () => {
    const wrapper = render(<Input type="textarea" rows={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('autoHeight', () => {
    jest.useFakeTimers();
    const props = {
      autoHeight: true,
      type: 'textarea',
      rows: 4,
      value: 'foo',
      onChange: jest.fn(),
    };
    const wrapper = mount(<Input {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('textarea').simulate('change', { target: { value: 'this is a test!' } });
    expect(props.onChange).toBeCalledWith('this is a test!');
    jest.runAllTimers();
    wrapper.unmount();
  });
});

describe('Input.Number', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="number" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders onClear called correctly', () => {
    const onClear = jest.fn();
    const wrapper = mount(
      <Input
        type="number"
        onClear={onClear}
        clearable
      />
    );

    const input = wrapper.find('input[type="hidden"]');
    input.simulate('change', { target: { value: 'My new value' } });
    wrapper.find('i.za-input-clear').simulate('click');
    expect(onClear).toHaveBeenCalled();
    expect(input.instance().value).toEqual('');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

//   it('enter number', () => {
//     const onChange = jest.fn();
//     const wrapper = mount(<Input type="number" onChange={onChange} />);
//     wrapper.find('input').simulate('focus');
//     const keys = wrapper.find('.za-keyboard-keys');
//     keys.childAt(0).simulate('click');
//     expect(onChange).toBeCalledWith('1');
//     wrapper.unmount();
//   });
});

describe('Input.Price', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="price" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Input.Idcard', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="idcard" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
