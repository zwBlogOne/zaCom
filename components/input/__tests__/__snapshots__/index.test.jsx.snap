// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  class="za-input za-input-text clearable"
>
  <input
    autocomplete="off"
    type="text"
    value=""
  />
  <i
    class="za-icon za-input-clear za-icon-wrong-round-fill"
  />
</div>
`;

exports[`Input renders onClear called correctly 1`] = `
<Input
  clearable={true}
  onClear={
    [MockFunction] {
      "calls": Array [
        Array [
          "My new value",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  type="text"
>
  <InputBase
    clearable={true}
    disabled={false}
    onClear={
      [MockFunction] {
        "calls": Array [
          Array [
            "My new value",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    prefixCls="za-input"
    type="text"
  >
    <div
      className="za-input za-input-text clearable"
    >
      <input
        autoComplete="off"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onCompositionUpdate={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <Icon
        className="za-input-clear"
        onClick={[Function]}
        prefixCls="za-icon"
        type="wrong-round-fill"
      >
        <i
          className="za-icon za-input-clear za-icon-wrong-round-fill"
          onClick={[Function]}
        />
      </Icon>
    </div>
  </InputBase>
</Input>
`;

exports[`Input showLength 1`] = `
<div
  class="za-input za-input-textarea"
>
  <textarea
    maxlength="100"
    rows="4"
  />
  <div
    class="za-input-length"
  >
    0/100
  </div>
</div>
`;

exports[`Input.Idcard renders correctly 1`] = `
<Input
  type="idcard"
>
  <InputNumber
    clearable={true}
    disabled={false}
    prefixCls="za-input"
    type="idcard"
  >
    <div
      className="za-input za-input-number clearable"
      onClick={[Function]}
    >
      <div
        className="za-input-placeholder"
      />
      <div
        className="za-input-content"
      />
      <input
        disabled={false}
        type="hidden"
        value=""
      />
      <KeyboardPicker
        onKeyClick={[Function]}
        prefixCls="za-keyboard"
        type="idcard"
        visible={false}
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={false}
          maskType="normal"
          prefixCls="za-popup"
          stayTime={3000}
          visible={false}
        >
          <Portal
            getContainer={[Function]}
          >
            <div
              className="za-popup za-popup-hidden"
            >
              <div
                className="za-popup-wrapper za-popup-wrapper-bottom"
                style={
                  Object {
                    "WebkitTransitionDuration": "200ms",
                    "transitionDuration": "200ms",
                  }
                }
              >
                <Keyboard
                  onKeyClick={[Function]}
                  prefixCls="za-keyboard"
                  type="idcard"
                  visible={false}
                >
                  <div
                    className="za-keyboard"
                    onClick={[Function]}
                  >
                    <div
                      className="za-keyboard-keys"
                    >
                      <div
                        className="za-keyboard-item"
                        key="0"
                        onClick={[Function]}
                      >
                        1
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="1"
                        onClick={[Function]}
                      >
                        2
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="2"
                        onClick={[Function]}
                      >
                        3
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="3"
                        onClick={[Function]}
                      >
                        4
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="4"
                        onClick={[Function]}
                      >
                        5
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="5"
                        onClick={[Function]}
                      >
                        6
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="6"
                        onClick={[Function]}
                      >
                        7
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="7"
                        onClick={[Function]}
                      >
                        8
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="8"
                        onClick={[Function]}
                      >
                        9
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="9"
                        onClick={[Function]}
                      >
                        x
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="10"
                        onClick={[Function]}
                      >
                        0
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="11"
                        onClick={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="keyboard"
                        >
                          <i
                            className="za-icon za-icon-keyboard"
                          />
                        </Icon>
                      </div>
                    </div>
                    <div
                      className="za-keyboard-handle"
                    >
                      <div
                        className="za-keyboard-item"
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="deletekey"
                        >
                          <i
                            className="za-icon za-icon-deletekey"
                          />
                        </Icon>
                      </div>
                      <div
                        className="za-keyboard-item za-keyboard-item-ok"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                  </div>
                </Keyboard>
              </div>
            </div>
          </Portal>
        </Popup>
      </KeyboardPicker>
      <Icon
        className="za-input-clear"
        onClick={[Function]}
        prefixCls="za-icon"
        type="wrong-round-fill"
      >
        <i
          className="za-icon za-input-clear za-icon-wrong-round-fill"
          onClick={[Function]}
        />
      </Icon>
    </div>
  </InputNumber>
</Input>
`;

exports[`Input.Number renders correctly 1`] = `
<Input
  type="number"
>
  <InputNumber
    clearable={true}
    disabled={false}
    prefixCls="za-input"
    type="number"
  >
    <div
      className="za-input za-input-number clearable"
      onClick={[Function]}
    >
      <div
        className="za-input-placeholder"
      />
      <div
        className="za-input-content"
      />
      <input
        disabled={false}
        type="hidden"
        value=""
      />
      <KeyboardPicker
        onKeyClick={[Function]}
        prefixCls="za-keyboard"
        type="number"
        visible={false}
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={false}
          maskType="normal"
          prefixCls="za-popup"
          stayTime={3000}
          visible={false}
        >
          <Portal
            getContainer={[Function]}
          >
            <div
              className="za-popup za-popup-hidden"
            >
              <div
                className="za-popup-wrapper za-popup-wrapper-bottom"
                style={
                  Object {
                    "WebkitTransitionDuration": "200ms",
                    "transitionDuration": "200ms",
                  }
                }
              >
                <Keyboard
                  onKeyClick={[Function]}
                  prefixCls="za-keyboard"
                  type="number"
                  visible={false}
                >
                  <div
                    className="za-keyboard"
                    onClick={[Function]}
                  >
                    <div
                      className="za-keyboard-keys"
                    >
                      <div
                        className="za-keyboard-item"
                        key="0"
                        onClick={[Function]}
                      >
                        1
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="1"
                        onClick={[Function]}
                      >
                        2
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="2"
                        onClick={[Function]}
                      >
                        3
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="3"
                        onClick={[Function]}
                      >
                        4
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="4"
                        onClick={[Function]}
                      >
                        5
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="5"
                        onClick={[Function]}
                      >
                        6
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="6"
                        onClick={[Function]}
                      >
                        7
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="7"
                        onClick={[Function]}
                      >
                        8
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="8"
                        onClick={[Function]}
                      >
                        9
                      </div>
                      <div
                        className="za-keyboard-item za-keyboard-item-disabled"
                        key="9"
                        onClick={[Function]}
                      />
                      <div
                        className="za-keyboard-item"
                        key="10"
                        onClick={[Function]}
                      >
                        0
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="11"
                        onClick={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="keyboard"
                        >
                          <i
                            className="za-icon za-icon-keyboard"
                          />
                        </Icon>
                      </div>
                    </div>
                    <div
                      className="za-keyboard-handle"
                    >
                      <div
                        className="za-keyboard-item"
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="deletekey"
                        >
                          <i
                            className="za-icon za-icon-deletekey"
                          />
                        </Icon>
                      </div>
                      <div
                        className="za-keyboard-item za-keyboard-item-ok"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                  </div>
                </Keyboard>
              </div>
            </div>
          </Portal>
        </Popup>
      </KeyboardPicker>
      <Icon
        className="za-input-clear"
        onClick={[Function]}
        prefixCls="za-icon"
        type="wrong-round-fill"
      >
        <i
          className="za-icon za-input-clear za-icon-wrong-round-fill"
          onClick={[Function]}
        />
      </Icon>
    </div>
  </InputNumber>
</Input>
`;

exports[`Input.Number renders onClear called correctly 1`] = `
<Input
  clearable={true}
  onClear={
    [MockFunction] {
      "calls": Array [
        Array [
          "",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  type="number"
>
  <InputNumber
    clearable={true}
    disabled={false}
    onClear={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    prefixCls="za-input"
    type="number"
  >
    <div
      className="za-input za-input-number focus clearable"
      onClick={[Function]}
    >
      <div
        className="za-input-placeholder"
      />
      <div
        className="za-input-content"
      />
      <input
        disabled={false}
        type="hidden"
        value=""
      />
      <KeyboardPicker
        onKeyClick={[Function]}
        prefixCls="za-keyboard"
        type="number"
        visible={true}
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={false}
          maskType="normal"
          prefixCls="za-popup"
          stayTime={3000}
          visible={true}
        >
          <Portal
            getContainer={[Function]}
          >
            <div
              className="za-popup"
            >
              <div
                className="za-popup-wrapper za-popup-wrapper-bottom"
                style={
                  Object {
                    "WebkitTransitionDuration": "200ms",
                    "transitionDuration": "200ms",
                  }
                }
              >
                <Keyboard
                  onKeyClick={[Function]}
                  prefixCls="za-keyboard"
                  type="number"
                  visible={true}
                >
                  <div
                    className="za-keyboard"
                    onClick={[Function]}
                  >
                    <div
                      className="za-keyboard-keys"
                    >
                      <div
                        className="za-keyboard-item"
                        key="0"
                        onClick={[Function]}
                      >
                        1
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="1"
                        onClick={[Function]}
                      >
                        2
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="2"
                        onClick={[Function]}
                      >
                        3
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="3"
                        onClick={[Function]}
                      >
                        4
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="4"
                        onClick={[Function]}
                      >
                        5
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="5"
                        onClick={[Function]}
                      >
                        6
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="6"
                        onClick={[Function]}
                      >
                        7
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="7"
                        onClick={[Function]}
                      >
                        8
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="8"
                        onClick={[Function]}
                      >
                        9
                      </div>
                      <div
                        className="za-keyboard-item za-keyboard-item-disabled"
                        key="9"
                        onClick={[Function]}
                      />
                      <div
                        className="za-keyboard-item"
                        key="10"
                        onClick={[Function]}
                      >
                        0
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="11"
                        onClick={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="keyboard"
                        >
                          <i
                            className="za-icon za-icon-keyboard"
                          />
                        </Icon>
                      </div>
                    </div>
                    <div
                      className="za-keyboard-handle"
                    >
                      <div
                        className="za-keyboard-item"
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="deletekey"
                        >
                          <i
                            className="za-icon za-icon-deletekey"
                          />
                        </Icon>
                      </div>
                      <div
                        className="za-keyboard-item za-keyboard-item-ok"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                  </div>
                </Keyboard>
              </div>
            </div>
          </Portal>
        </Popup>
      </KeyboardPicker>
      <Icon
        className="za-input-clear"
        onClick={[Function]}
        prefixCls="za-icon"
        type="wrong-round-fill"
      >
        <i
          className="za-icon za-input-clear za-icon-wrong-round-fill"
          onClick={[Function]}
        />
      </Icon>
    </div>
  </InputNumber>
</Input>
`;

exports[`Input.Price renders correctly 1`] = `
<Input
  type="price"
>
  <InputNumber
    clearable={true}
    disabled={false}
    prefixCls="za-input"
    type="price"
  >
    <div
      className="za-input za-input-number clearable"
      onClick={[Function]}
    >
      <div
        className="za-input-placeholder"
      />
      <div
        className="za-input-content"
      />
      <input
        disabled={false}
        type="hidden"
        value=""
      />
      <KeyboardPicker
        onKeyClick={[Function]}
        prefixCls="za-keyboard"
        type="price"
        visible={false}
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={false}
          maskType="normal"
          prefixCls="za-popup"
          stayTime={3000}
          visible={false}
        >
          <Portal
            getContainer={[Function]}
          >
            <div
              className="za-popup za-popup-hidden"
            >
              <div
                className="za-popup-wrapper za-popup-wrapper-bottom"
                style={
                  Object {
                    "WebkitTransitionDuration": "200ms",
                    "transitionDuration": "200ms",
                  }
                }
              >
                <Keyboard
                  onKeyClick={[Function]}
                  prefixCls="za-keyboard"
                  type="price"
                  visible={false}
                >
                  <div
                    className="za-keyboard"
                    onClick={[Function]}
                  >
                    <div
                      className="za-keyboard-keys"
                    >
                      <div
                        className="za-keyboard-item"
                        key="0"
                        onClick={[Function]}
                      >
                        1
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="1"
                        onClick={[Function]}
                      >
                        2
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="2"
                        onClick={[Function]}
                      >
                        3
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="3"
                        onClick={[Function]}
                      >
                        4
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="4"
                        onClick={[Function]}
                      >
                        5
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="5"
                        onClick={[Function]}
                      >
                        6
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="6"
                        onClick={[Function]}
                      >
                        7
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="7"
                        onClick={[Function]}
                      >
                        8
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="8"
                        onClick={[Function]}
                      >
                        9
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="9"
                        onClick={[Function]}
                      >
                        .
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="10"
                        onClick={[Function]}
                      >
                        0
                      </div>
                      <div
                        className="za-keyboard-item"
                        key="11"
                        onClick={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="keyboard"
                        >
                          <i
                            className="za-icon za-icon-keyboard"
                          />
                        </Icon>
                      </div>
                    </div>
                    <div
                      className="za-keyboard-handle"
                    >
                      <div
                        className="za-keyboard-item"
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                      >
                        <Icon
                          prefixCls="za-icon"
                          type="deletekey"
                        >
                          <i
                            className="za-icon za-icon-deletekey"
                          />
                        </Icon>
                      </div>
                      <div
                        className="za-keyboard-item za-keyboard-item-ok"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                  </div>
                </Keyboard>
              </div>
            </div>
          </Portal>
        </Popup>
      </KeyboardPicker>
      <Icon
        className="za-input-clear"
        onClick={[Function]}
        prefixCls="za-icon"
        type="wrong-round-fill"
      >
        <i
          className="za-icon za-input-clear za-icon-wrong-round-fill"
          onClick={[Function]}
        />
      </Icon>
    </div>
  </InputNumber>
</Input>
`;

exports[`Input.Textarea autoHeight 1`] = `
<Input
  autoHeight={true}
  onChange={[MockFunction]}
  rows={4}
  type="textarea"
  value="foo"
>
  <InputTextarea
    autoHeight={true}
    clearable={false}
    disabled={false}
    focused={false}
    onChange={[MockFunction]}
    prefixCls="za-input"
    rows={4}
    showLength={false}
    value="foo"
  >
    <div
      className="za-input za-input-textarea"
    >
      <textarea
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        rows={4}
        value="foo"
      />
    </div>
  </InputTextarea>
</Input>
`;

exports[`Input.Textarea renders correctly 1`] = `
<div
  class="za-input za-input-textarea"
>
  <textarea
    rows="4"
  />
</div>
`;
